### PR TITLE
Addition to CBMC infrastructure

### PR DESCRIPTION
**Pull request by Bujain**
*Description of changes:*
These files are modified such that they can be directly used as the template for the proofs. The files which are modified are
.gitmodules
Added .gitignore
Makefile.common (added include paths and preprocessor definitions)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -11,6 +11,8 @@
 [submodule "test/cbmc/aws-templates-for-cbmc-proofs"]
 	path = test/cbmc/aws-templates-for-cbmc-proofs
 	url = https://github.com/awslabs/aws-templates-for-cbmc-proofs.git
+	update = none
 [submodule "test/cbmc/litani"]
 	path = test/cbmc/litani
 	url = https://github.com/awslabs/aws-build-accumulator.git
+	update = none
```
Comment (AniruddhaKanhere) [inline .gitmodules]: Add an empty line here

File: test/cbmc/proofs/Makefile-template-defines [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Absolute path to the litani script.
+#
+LITANI ?= $(abspath $(PROOF_ROOT)/../litani/litani)
+
+
+# Name of this proof project, displayed in proof reports. For example,
+# "s2n" or "Amazon FreeRTOS". For projects with multiple proof roots,
+# this may be overridden on the command-line to Make, for example
+#
+# 	  make PROJECT_NAME="FreeRTOS MQTT" report
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/Makefile-template-defines:16]: While this is a perfectly fine example, I would say change it to below:
```suggestion
# 	  make PROJECT_NAME="AWS IoT Over-the-air Update Library" report
```

File: .gitmodules [hunk 3]
```diff
@@ -11,6 +11,7 @@
 [submodule "test/cbmc/aws-templates-for-cbmc-proofs"]
 	path = test/cbmc/aws-templates-for-cbmc-proofs
 	url = https://github.com/awslabs/aws-templates-for-cbmc-proofs.git
+	update = none
 [submodule "test/cbmc/litani"]
 	path = test/cbmc/litani
 	url = https://github.com/awslabs/aws-build-accumulator.git
```
Comment (AniruddhaKanhere) [inline .gitmodules:17]: We should add an `update = none` here.

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 3 above)
Comment (Bujain) [reply .gitmodules:17]: I have added the comment in the last commit

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Pull request merged**